### PR TITLE
Add case for random endpoint when no available unreviewed items

### PR DESCRIPTION
**Pull request by guergana**
Bug: [T302289](https://phabricator.wikimedia.org/T302289)

### CODE REVIEW

**Review by lucaswerkmeister — COMMENTED** [review]
I tried this out locally, and it doesn’t seem to quite work yet – when I go to `/random` when there are no mismatches, I see an empty results page, but the “no mismatches available” message doesn’t show up. So far I can’t figure out why.

I’m also trying to understand how this works (not being familiar with Laravel yet)… the call to `inertia('Results', [ 'user' => $user ])` in `RandomizeController` directly renders `Results.vue`, right? Or does it go through `ResultsController`? (My thinking is that, if it bypasses `ResultsController`, then we shouldn’t need to change the code in there. But maybe I’m misunderstanding something.)

**Comment by guergana** [comment]
> I tried this out locally, and it doesn’t seem to quite work yet – when I go to `/random` when there are no mismatches, I see an empty results page, but the “no mismatches available” message doesn’t show up. So far I can’t figure out why.
> 
> I’m also trying to understand how this works (not being familiar with Laravel yet)… the call to `inertia('Results', [ 'user' => $user ])` in `RandomizeController` directly renders `Results.vue`, right? Or does it go through `ResultsController`? (My thinking is that, if it bypasses `ResultsController`, then we shouldn’t need to change the code in there. But maybe I’m misunderstanding something.)

You are right! Those are leftovers from the rebasing. Sorry about that. Will upload a new version.

**Comment by lucaswerkmeister** [comment]
> I tried this out locally, and it doesn’t seem to quite work yet – when I go to `/random` when there are no mismatches, I see an empty results page, but the “no mismatches available” message doesn’t show up. So far I can’t figure out why.

It turns out I needed to run `vendor/bin/sail npm run watch` (or `dev` instead of `watch`) to update the front-end files.

File: resources/js/Pages/Results.vue [hunk 1]
```diff
@@ -61,6 +61,9 @@
                     </wikit-link>
                 </span>
             </Message>
+            <Message type="notice" v-if="item_ids.length === 0">
```
Comment (lucaswerkmeister) [inline resources/js/Pages/Results.vue:64]: Maybe a comment around this message would make sense? Something like:

```html
<!-- The Results page without item_ids is used by RandomizeController. -->
```

**Review by lucaswerkmeister — APPROVED** [review]
Looks pretty good :+1: one optional suggestion.

File: resources/js/Pages/Results.vue (hunk 1 above)
Comment (guergana) [reply resources/js/Pages/Results.vue:64]: ok. i will add it.

Comment (guergana) [reply resources/js/Pages/Results.vue:65]: Done.

**Review by lucaswerkmeister — APPROVED** [review]
The test looks really nice :+1:

**Pull request merged**